eng/scripts/appium-install.ps1

```diff
@@ -40,27 +40,49 @@ Find the script for that on the DevDiv Azure DevOps instance, Engineering team,
 
 param
 (
-    [string] $appiumVersion = '2.1.1',
-    [string] $windowsDriverVersion = '2.10.1',
-    [string] $androidDriverVersion = '2.29.4',
-    [string] $iOSDriverVersion = '5.2.0',
-    [string] $macDriverVersion = '1.7.2'
+    [string] $appiumVersion = '2.5.4',
+    [string] $windowsDriverVersion = '2.12.23',
+    [string] $androidDriverVersion = '3.5.1',
+    [string] $iOSDriverVersion = '7.16.1',
+    [string] $macDriverVersion = '1.17.3'
 )
 
 Write-Output  "Welcome to the Appium installer"
 
 Write-Output  "Node version"
 node -v
 
-Write-Output  "Installing appium $appiumVersion"
-npm install -g appium@$appiumVersion
+# If there's a ~/.appium folder, remove it as it can cause issues from v1
+# it might also generally have caching issues for us with our runs
+$appiumUserData = "$env:USERPROFILE/.appium"
+if (Test-Path $appiumUserData) {
+    Write-Output  "Removing $appiumUserData"
+    Remove-Item -Path $appiumUserData -Force -Recurse
+}
 
-write-Output  "Installed appium"
-appium -v
+# Check for an existing appium install version
+$appiumCurrentVersion = ""
+try { $appiumCurrentVersion = appium -v | Out-String } catch { }
 
-Write-Output  "Installing appium doctor"
-npm install -g appium-doctor
+if ($appiumCurrentVersion) {
+    Write-Output  "Existing Appium version $appiumCurrentVersion"
+} else {
+    Write-Output  "No Appium version installed"
+}
+
+# If current version does not match the one we want, uninstall and install the new version
+if ($appiumCurrentVersion -ne $appiumVersion) {
+    Write-Output  "Uninstalling appium $appiumCurrentVersion"
+    npm uninstall -g appium
+    Write-Output  "Uninstalled appium $appiumCurrentVersion"
 
+    Write-Output  "Installing appium $appiumVersion"
+    npm install -g appium@$appiumVersion
+    write-Output  "Installed appium $appiumVersion"   
+}
+
+Write-Output  "Installing appium doctor"
+npm install -g @appium/doctor
```

Comment:
More of an FYI, we should remove this at some point as Appium doctor is deprecated

Reply:
Ahh yeah i misread the initial deprecation notice and thought it was just a new package... I see it's now baked into appium itself... Updated!